harmony/http.py

```diff
@@ -260,18 +169,53 @@ def _download(config, url: str, access_token: str, data, total_retries: int, use
     if user_agent is not None:
         headers['user-agent'] = user_agent
     auth = EarthdataAuth(config.oauth_uid, config.oauth_password, access_token)
-    with _mount_retry(_earthdata_session(), total_retries) as session:
-        session.auth = auth
-        if data is None:
-            return session.get(url, headers=headers, timeout=TIMEOUT, **kwargs_download_agent)
-        else:
-            # Including this header since the stdlib does by default,
-            # but we've switched to `requests` which does not.
-            headers['Content-Type'] = 'application/x-www-form-urlencoded'
-            return session.post(url, headers=headers, data=data, timeout=TIMEOUT)
-
-
-def _download_with_fallback_authn(config, url: str, data, total_retries: int, user_agent=None, **kwargs_download_agent):
+    tries = 0
+    retry = True
+    response = None
+    while retry is True:
```

Comment:
It could but for some reason I've been brainwashed to consider `while True` to be a bad idea. I kind of think of that check as a guardrail in case something inside the loop changes.